internal/external-dns/registry/types.go

```diff
@@ -0,0 +1,14 @@
+package registry
+
+import "sigs.k8s.io/external-dns/endpoint"
+
+type nameMapper interface {
+	toEndpointName(txtDNSName string) (endpointName, recordType, ownerID string)
+	toTXTName(string, string, string) string
```

Comment:
this feels a little specific to TXT records, should this be: `toRegistryKey` or something like that?

Reply:
Moved back